File: client/lib/models/responses/meal_item.dart
import 'food.dart';
import 'macros.dart';

/// One item inside a `DietMeal` — food + amount/unit + scaled macros.
class MealItem {
  MealItem({
    required this.id,
    required this.position,
    required this.amount,
    required this.unit,
    required this.food,
    required this.macros,
  });

  final int id;
  final int position;
  final double amount;
  final String unit;
  final Food? food;
  final Macros macros;

  factory MealItem.fromJson(Map<String, dynamic> json) {
    return MealItem(
      id: (json['id'] as num).toInt(),
      position: (json['position'] as num).toInt(),
      amount: (json['amount'] as num).toDouble(),
      unit: json['unit'] as String,
      food: json['food'] == null
          ? null
          : Food.fromJson(json['food'] as Map<String, dynamic>),
      macros: Macros.fromJson(json['macros'] as Map<String, dynamic>),
    );
  }
}

File: api/src/main/java/com/example/athletehub/model/MealItem.java
package com.example.athletehub.model;

import jakarta.persistence.*;
import lombok.*;

import java.math.BigDecimal;

/**
 * One food entry inside a {@link DietMeal} — target amount + unit +
 * position. Unit values are constrained by the schema CHECK to
 * {@code g | ml | portion}.
 */
@Entity
@Table(name = "meal_items")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class MealItem {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "meal_id", nullable = false)
    private Long mealId;

    @Column(name = "food_id", nullable = false)
    private Long foodId;

    @Column(nullable = false)
    private BigDecimal amount;

    @Column(nullable = false)
    private String unit;

    @Column(nullable = false)
    private int position;
}
